src/marqo/tensor_search/models/index_settings.py

```diff
@@ -250,6 +250,19 @@ def from_marqo_index(cls, marqo_index: core.MarqoIndex) -> "IndexSettings":
         else:
             raise api_exceptions.InternalError(f"Unknown index type: {type(marqo_index)}")
 
+    @classmethod
+    def get_model_properties(cls, marqo_index):
+        if marqo_index.model.properties is None:
+            return None
+
+        if marqo_index.model.properties.get('isMarqtuneModel', False):
+            # Hide all properties except for isMarqtuneModel
+            marqo_index.model.properties.pop('name', None)
+            marqo_index.model.properties.pop('dimensions')
+            marqo_index.model.properties.pop('model_location')
+            marqo_index.model.properties.pop('type')
```

Comment:
we only provide a default value for 'name', does that mean all other properties always has value? is it safer to provide default values for everyone of them to avoid KeyError?

Reply:
Only name is optional as it wont exist for hf models, all others are mandatory.